pkg/apis/flowcontrol/types.go

```diff
@@ -0,0 +1,298 @@
+	// Required.
+	APIGroups []string
+	// APIVersions is the API versions the resources belong to. '*' is all versions.
+	// If '*' is present, the length of the slice must be one.
+	// Required.
+	APIVersions []string
+	// Resources is a list of resources this rule applies to.
+	//
+	// For example:
+	// 'pods' means pods.
+	// 'pods/log' means the log subresource of pods.
+	// '*' means all resources, but not subresources.
+	// 'pods/*' means all subresources of pods.
+	// '*/scale' means all scale subresources.
+	// '*/*' means all resources and their subresources.
+	//
+	// If wildcard is present, the validation rule will ensure resources do not
+	// overlap with each other.
+	//
+	// Depending on the enclosing object, subresources might not be allowed.
+	// Required.
+	Resources []string
+	// Verbs is a list of verbs the rule applies to. An empty set means that everything is allowed.
+	// +optional
+	Verbs []string
+	// ResourceNames is an optional white list of names that the rule applies to.  An empty set means that everything is allowed.
+	// +optional
+	ResourceNames []string
+	// NonResourceURLs is a set of partial urls that a user should have access to.  *s are allowed, but only as the full, final step in the path
+	// If an action is not a resource API request, then the URL is split on '/' and is checked against the NonResourceURLs to look for a match.
+	// Since non-resource URLs are not namespaced, this field is only applicable for ClusterRoles referenced from a ClusterRoleBinding.
+	// Rules can either apply to API resources (such as "pods" or "secrets") or non-resource URL paths (such as "/api"),  but not both.
+	// +optional
+	NonResourceURLs []string
+}
+
+// FlowSchemaStatus represents the current state of a flow-schema.
+type FlowSchemaStatus struct {
+	// Current state of flow-schema.
+	Conditions []FlowSchemaStatusCondition
+}
+
+// FlowSchemaStatusCondition describes conditions for a flow-schema.
+type FlowSchemaStatusCondition struct {
+	// Type is the type of the condition.
+	Type FlowSchemaStatusConditionType
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status ConditionStatus
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string
+	// Human-readable message indicating details about last transition.
+	Message string
+}
+
+// FlowSchemaStatusConditionType is a valid value for FlowSchemaStatusCondition.Type
+type FlowSchemaStatusConditionType string
+
+// FlowSchemaStatusConditionStatus is a valid value for FlowSchemaStatusCondition.Status
+type FlowSchemaStatusConditionStatus string
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RequestPriority represents the configuration of a priority level.
+type RequestPriority struct {
```

Comment:
PriorityLevelConfiguration?

Reply:
i in favor of this name. sounds clearer than `RequestPriority`.

Reply:
I am not in favor of adding the word "configuration" to this type name.  Why is this one type more of a configuration than any other here?  I do not think the word "configuration" is adding enough information to be worth its verbosity.  What is wrong with "RequestPriority"?

Reply:
what say @lavalamp ?

Reply:
We have stopped debating this, so I think we can mark this conversation resolved.